### PR TITLE
Update xarray geopandas

### LINKED ISSUE #925: Make code compatible with newest version of xarray
### Kind of request

Changing existing functionality

### Enhancement Description

The xarray version is now pinned at v2024.3.0 because of breaking in changes in latest version of xarray (v2024.05.0). 
Using the latest version results in these errors:
image

### Use case

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Tjalling-dejong**
## Issue addressed
Fixes #925 and makes the code compatible with GeoPandas 1.0.0

## Explanation
Changes for using GeoPandas 1.0.0:
- `unary_union` to  `union_all()`
- removed mode from gpd.read_file(), since this is only applicable for the fiona backend of geopandas. With version 1.0.0 GeoPandas now uses pyogrio by default.
- bbox argument of `gpd.read_file()` can now only be a tuple. I added a check if the bounding box is a list and added code to convert the bounding box to tuple if that is the case.
- In the process I also added typing to `io.readers.open_vector()`

For xarray I had to change the update_geometry function. The assigning of geometry coordinates broke due to xarray converting a `GeoDataArray` to an xarray pandas extension array. The GeoDataArray is now converted to a numpy array of geometries to tackle this bug. 

## General Checklist
- [ ] Updated tests or added new tests
- [ ] Branch is up to date with `main`
- [ ] Tests & pre-commit hooks pass
- [ ] Updated documentation
- [ ] Updated changelog.rst

## Data/Catalog checklist
- [ ] `data/catalogs/predefined_catalogs.yml` has not been modified.
- [ ] None of the old `data_catalog.yml` files have been chagned
- [ ] `data/chagnelog.rst` has been updated
- [ ] new file uses `LF` line endings (done automatically if you used `update_versions.py`)
- [ ] New file has been tested locally
- [ ] Tests have been added using the new file in the test suite

## Additional Notes (optional)
Add any additional notes or information that may be helpful.

### CODE REVIEW

File: hydromt/io/readers.py
Comment (savente93) [inline hydromt/io/readers.py:593]: Since #1011 we don't pass around loggers anymore, see migration guide for updates.

**Review by savente93 — CHANGES_REQUESTED** [review]
see comment about passing around loggers. Also please update the description to give a brief overview of what you changed. even if it's just "`unary_union` is now `union_all()`"

File: hydromt/io/readers.py
Comment (Tjalling-dejong) [reply hydromt/io/readers.py]: I removed it

**Review by savente93 — COMMENTED** [review]
Thanks for picking this up Tjalling. I just merged a fairly big renaming PR, that caused some conflicts here. Sorry about that, but the conflicts are much easier to manage on this side, so hope you don't mind. The PR looks good, and I'll be happy to approve once you fix the merge conflicts (or if you want I can do that)

**Comment by Tjalling-dejong** [comment]
Thanks for the review @savente93!

**Review by savente93 — APPROVED** [review]
All good now!

**Pull request merged**
